game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -102,49 +105,51 @@ public JComponent getUiComponent() {
-        if (value.isEmpty()) {
-          return false;
-        }
-
-        try {
-          return Integer.parseInt(value) > 0;
-        } catch (final NumberFormatException e) {
-          return false;
-        }
-      }
-
-      @Override
-      public String validValueDescription() {
-        return "Proxy host can be a network name or an IP address, port should be number, usually 4 to 5 digits.";
-      }
-
       @Override
       public void save(final SaveContext context) {
         if (noneButton.isSelected()) {
           context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.NONE);
+          context.setValue(proxyHostClientSetting, null);
+          context.setValue(proxyPortClientSetting, null);
         } else if (systemButton.isSelected()) {
           context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_SYSTEM_SETTINGS);
+          context.setValue(proxyHostClientSetting, null);
+          context.setValue(proxyPortClientSetting, null);
           HttpProxy.updateSystemProxy();
         } else {
-          context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_USER_PREFERENCES);
+          final String encodedHost = hostText.getText().trim();
+          final Optional<String> optionalHost = parseHost(encodedHost);
+          OptionalUtils.ifEmpty(optionalHost, () -> context.reportError(
+              proxyHostClientSetting,
+              "must be a network name or an IP address",
+              encodedHost));
+
+          final String encodedPort = portText.getText().trim();
+          final Optional<Integer> optionalPort = parsePort(encodedPort);
+          OptionalUtils.ifEmpty(optionalPort, () -> context.reportError(
+              proxyPortClientSetting,
+              "must be a positive integer, usually 4 to 5 digits",
+              encodedPort));
+
+          OptionalUtils.ifAllPresent(optionalHost, optionalPort, (host, port) -> {
+            context.setValue(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_USER_PREFERENCES);
+            context.setValue(proxyHostClientSetting, host);
+            context.setValue(proxyPortClientSetting, port);
+          });
         }
+      }
 
-        final String host = hostText.getText().trim();
-        context.setValue(proxyHostClientSetting, host.isEmpty() ? null : host);
+      private Optional<String> parseHost(final String encodedHost) {
+        return !encodedHost.isEmpty() ? Optional.of(encodedHost) : Optional.empty();
+      }
 
-        final String encodedPort = portText.getText().trim();
-        context.setValue(proxyPortClientSetting, encodedPort.isEmpty() ? null : Integer.valueOf(encodedPort));
+      private Optional<Integer> parsePort(final String encodedPort) {
+        try {
+          final Integer port = Integer.valueOf(encodedPort);
+          return (port > 0) ? Optional.of(port) : Optional.empty();
+        } catch (final NumberFormatException e) {
+          return Optional.empty();
```

Comment:
side-note, ideally we would not wait until a user hits save before telling them about form errors. Some ideas for future improvements:
- show a tooltip text briefly after user closes the save-action error message. This will give a visual cue of which forms need to be updated
- on focus lost, do validation, if invalid change text area foreground or background to red, show a tooltip with error message. On focus gained, restore the background color to default.